packages/backend/src/services/AsyncQueryService/AsyncQueryService.ts

```
@@ -1657,21 +1686,22 @@ export class AsyncQueryService extends ProjectService {
         const queryTags: RunQueryTags = {
             organization_uuid: organizationUuid,
             project_uuid: projectUuid,
-            user_uuid: user.userUuid,
+            user_uuid: user?.userUuid ?? '',
             explore_name: metricQuery.exploreName,
             query_context: context,
         };
 
-        const explore = await this.getExplore(
+        const explore = await this.getExplore({
             user,
+            account,
             projectUuid,
-            metricQuery.exploreName,
+            exploreName: metricQuery.exploreName,
             organizationUuid,
-        );
+        });
 
         const warehouseConnection = await this._getWarehouseClient(
             projectUuid,
-            await this.getWarehouseCredentials(projectUuid, user.userUuid),
+            await this.getWarehouseCredentials(projectUuid, user?.userUuid),
```

@almeidabbm there's nothing for us to pass here if we don't have a real user right?

---

This should be fine, perhaps the only change I would do is in `getWarehouseCredentials` instead of passing `user?.userUuid ?? ''` when user credentials are required we should throw an error saying that there's no user in that case